### DISCUSSION TITLE
Ideas? Wishes?

### DISCUSSION THREAD

**Discussion by jkriege2** [jkriege2/JKQtPlotter#121] [Ideas]
Dear all!

I'm looking for input, what  the users of JKQTPlotter (might) need interms of new features, graphs, ...

There is a list of ToDos, that I think, would do the library some good here:
https://jkriege2.github.io/JKQtPlotter/page_todo.html

But the question is also: 
What do you think?
What are your requirements?

**Comment by Snolandia** [comment]
Have you ever thought of adding 3d surface graphs?

**Reply by jkriege2** [reply]
Hi!
sorry for the delaed answer (RL gets in the way constantly ;-)
In short: No, I don't have plans to support 3D plots ... sorry

**Comment by Snolandia** [comment]
Also, adding the ability to click symbols and drag to change the data, as well as a slot/signal for symbol clicked, and hover over symbol changes the cursor. I implemented it on my thing, but it is slower performing than if it was implemented directly on the class and its objects. 

And maybe more slots/signals for everything? Would be nice to have more signal options.

**Reply by jkriege2** [reply]
- Which signals and slots would be of interest ... "everthing" is a bit non-specific and thus hard to provide ;-) ?
- There is a possibilit to get a signal, when the mouse hovers over a datapoint: [JKQtPlotter::tooltipDisplayed()](https://jkriege2.github.io/JKQtPlotter/class_j_k_q_t_plotter.html#a748601cd74cdb3722acdb9fb036fad58) ... it is called, when JKQTPMouseDragActions::jkqtpmdaToolTipForClosestDataPoint or JKQTPMouseMoveActions::jkqtpmmaToolTipForClosestDataPoint is activated and the mouse is near a datapoint ... but that also "just" uses [JKQTPPlotElement::hitTest()](https://jkriege2.github.io/JKQtPlotter/class_j_k_q_t_p_plot_element.html#a122b7b717c3c578861005aa8db8605e0) on all visible graphs to find the closest points (which might be quite slow) ... so you could use the same approach in an clicked-signal, e.g. [JKQTPlotter::plotMouseClicked()](https://jkriege2.github.io/JKQtPlotter/class_j_k_q_t_plotter.html#aa9a7e0dcbba89a0095ae62183dcbaea2)

What tpe of mode would ou like to have? I could think of adding a mode (which ou would have to activate manuall for performance reasons) that gives you a list of close graphs, when the mouse is clicked ... would that halp?

**Reply by Snolandia** [reply]
Totally missed this reply. Since im currently digging into the plotter pretty far, ill probably just add stuff as i go as a fork and let you decide on if it should be added.

**Comment by Social-Mean** [comment]
Does this library support polar plot? I have not seen any information about polar plot in documents.

**Reply by jkriege2** [reply]
Hi!
sorry for the delaed answer (RL gets in the way constantly ;-)
In short: No, not at the moment ... it seems more viable than 3D but still, currently I don't have plans for such extensions ...

**Comment by rishiiiiiiiiii10** [comment]
there should be a way to remove/delete the secondary axies from the base plotter
